memcached.c

```diff
@@ -3359,7 +3371,12 @@ static inline void process_get_command(conn *c, token_t *tokens, size_t ntokens,
                 return;
             }
 
-            it = limited_get(key, nkey, c);
+            if (should_touch) {
+                it = item_touch(key, nkey, exptime, c);
```

Comment:
this should move into limited_get, of which this function is the only caller. Then you don't lose the refcount protection by calling GAT instead of GET.

Reply:
@dormando 
I will make these changes soon. I am trying to follow the code paths, there seems to be a missing item_remove(it) in following path. Is that intentional? 

if ((it->it_flags & ITEM_CHUNKED) == 0)

https://github.com/memcached/memcached/blob/69a4414178e1bd5f69db7810022a94799d7e1019/memcached.c#L3404

Reply:
You mean there's a missing check to see if add_iov() fails if the item isn't chunked?

That's fine there (the iov's tends to get doubled, and it probably never actually fails). I'm going to refactor the loop soon so there's one iov space check at the top of the loop. That should fix it as well.